commit cb1d53549b9bc320f7ed2240276ef058817c1c0a
Author:     Yuhong Wen <yh_wen@hotmail.com>
AuthorDate: Mon May 16 19:54:05 2022 -0400
Commit:     GitHub <noreply@github.com>
CommitDate: Mon May 16 19:54:05 2022 -0400

    Fixed the issue abort_job command could not abort all running jobs. (#549)
---
 nvflare/private/fed/server/server_engine.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index 68bea5be..22e19e9c 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -335,11 +335,8 @@ class ServerEngine(ServerEngineInternalSpec):
         return ""
 
     def abort_app_on_server(self, run_number: str) -> str:
-        status = self.engine_info.status
-        if status == MachineStatus.STOPPED:
+        if run_number not in self.run_processes.keys():
             return "Server app has not started."
-        if status == MachineStatus.STARTING:
-            return "Server app is starting, please wait for started before abort."
 
         self.logger.info("Abort the server app run.")
 
